File: session-service-java/src/main/java/org/example/sessionservice/SessionController.java
package org.example.sessionservice;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/sessions")
@RequiredArgsConstructor
public class SessionController {

	private final SessionService sessionService;

	@PostMapping("/users/{userId}")
	public ResponseEntity<Session> createSession(@PathVariable String userId) {
		final Session session = new Session(userId);

		try {
			sessionService.saveSession(session);
		} catch (SessionService.SessionExistsException e) {
			return ResponseEntity.status(HttpStatus.CONFLICT).build();
		}
		return ResponseEntity.ok(session);
	}

	@GetMapping("/{id}")
	public ResponseEntity<Session> getSession(@PathVariable String id) {
		return sessionService.getSession(id)
				.map(ResponseEntity::ok)
				.orElseGet(() -> ResponseEntity.notFound().build());
	}
}

File: session-service/src/session/session.controller.ts
import { Controller, Post, Get, Param, HttpException, HttpStatus } from '@nestjs/common';
import { SessionService } from './session.service';
import { Session } from './session';

@Controller('sessions')
export class SessionController {
    constructor(private readonly sessionService: SessionService) {}

    @Post('users/:userId')
    async createSession(@Param('userId') userId: string): Promise<Session> {
        const session = new Session(userId);
        try {
            await this.sessionService.saveSession(session);
        } catch (e) {
            if (e.message === 'SessionExistsException') {
                throw new HttpException('Conflict', HttpStatus.CONFLICT);
            }
            throw e;
        }
        return session;
    }

    @Get(':id')
    async getSession(@Param('id') id: string): Promise<Session> {
        const session = await this.sessionService.getSession(id);
        if (!session) {
            throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
        }
        return session;
    }
}